test/Api.Test/Tools/Validators/SendRotationValidatorTests.cs

```diff
@@ -0,0 +1,130 @@
+
+        var sut = new SendRotationValidator(
+            sendService,
+            sendRepository
+        );
+
+        var user = new User { Id = new Guid() };
+        var sends = CreateSends();
+        var userSends = sends.Select(s =>
+            new Send { Id = s.Id.Value, Type = s.Type, Data = "{}" }).ToList();
+        userSends.Add(new Send { Id = Guid.NewGuid(), Data = "{}" });
+        sendRepository.GetManyByUserIdAsync(user.Id).Returns(userSends);
+
+        // Act, Assert
+        await Assert.ThrowsAsync<BadRequestException>(async () => await sut.ValidateAsync(user, sends));
+    }
+
+    [Fact]
+    public async Task ValidateAsync_SendDoesNotBelongToUser_SendNotReturned()
+    {
+        // Arrange
+        var sendService = Substitute.For<ISendService>();
+        var sendRepository = Substitute.For<ISendRepository>();
+
+        var sut = new SendRotationValidator(
+            sendService,
+            sendRepository
+        );
+
+        var user = new User { Id = new Guid() };
+        var sends = CreateSends();
+        var userSends = sends.Select(s =>
+            new Send { Id = s.Id.Value, Type = s.Type, Data = "{}" }).ToList();
+        userSends.RemoveAt(0);
+        sendRepository.GetManyByUserIdAsync(user.Id).Returns(userSends);
+
+        var result = await sut.ValidateAsync(user, sends);
+
+        Assert.DoesNotContain(result, c => c.Id == sends.First().Id);
+    }
+
+    private IEnumerable<SendWithIdRequestModel> CreateSends()
+    {
+        return new[]
+        {
+            new SendWithIdRequestModel
+            {
+                DeletionDate = DateTime.UtcNow.AddDays(20),
+                Disabled = false,
+                Id = new Guid("72e9ac6d-05f4-4227-ae0d-8a5207623a1a"),
+                Key = "Send1Key",
+                Name = "Send 1",
+                Type = SendType.Text,
+                Text = new SendTextModel(new SendTextData("Text name", "Notes", "Encrypted text", false))
+            },
+            new SendWithIdRequestModel
+            {
+                DeletionDate = DateTime.UtcNow.AddDays(20),
+                Disabled = true,
+                Id = new Guid("6b55836c-9280-4589-8762-01b0d8172c97"),
+                Key = "Send2Key",
+                Name = "Send 2",
+                Type = SendType.Text,
+                Text = new SendTextModel(new SendTextData("Text name", "Notes", "Encrypted text for Send 2",
+                    false)),
+            },
+            new SendWithIdRequestModel
+            {
+                DeletionDate = DateTime.UtcNow.AddDays(20),
+                Disabled = false,
+                Id = new Guid("9a65bbfb-8138-4aa5-a572-e5c0a41b540e"),
+                Key = "Send3Key",
+                Name = "Send 3",
+                Type = SendType.File,
+                File = new SendFileModel(new SendFileData("File name", "Notes", "File name here")),
+                HideEmail = true
+            }
```

Comment:
📝  This data set is being used as a test input, to derive the mock output, and in one case to derive the expected result of the test. These three cases should be treated separately. More precise guidance for each case is provided above.